### PR TITLE
Support for missing endpoints + rename to better fit documentation

### PR DESCRIPTION
**Pull request by swiftyspiffy**
- rename PutUsersAsync to UpdateUsersAsync
- rename UpdateCustomRewardRedemptionStatus to UpdateRedemptionStatus
- adds support for GetExtensionLiveChannels
- add support for GetReleasedExtensions
- add support for GetExtensionBitsProducts
- add support for UpdateExtensionBitsProduct
- add support for GetChanneliCalendar
- add support for GetFollowedStreams
- rename GetStreamMarker to GetStreamMarkers

### CODE REVIEW

File: TwitchLib.Api.Helix/Extensions.cs [hunk 1]
```diff
@@ -40,5 +41,28 @@ public Task<GetExtensionTransactionsResponse> GetExtensionTransactionsAsync(stri
         }
 
         #endregion
+
+        #region GetExtensionLiveChannels
+
+        public Task<GetExtensionLiveChannelsResponse> GetExtensionLiveChannelsAsync(string extensionId, int first = 20, string after = null, string accessTokenOrApplicationAccessToken = null)
```
Comment (Syzuna) [inline TwitchLib.Api.Helix/Extensions.cs:48]: why not just access token? I mean I get the intention but it makes the variable name way too long imo and we just merged your PR that unified that variable name to accessToken

File: TwitchLib.Api.Helix.Models/Extensions/ReleaseExtensions/Component.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
+﻿using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace TwitchLib.Api.Helix.Models.Extensions.ReleaseExtensions
```
Comment (Syzuna) [inline TwitchLib.Api.Helix.Models/Extensions/ReleaseExtensions/Component.cs:6]: the namespace should be ReleasedExtensions

File: TwitchLib.Api.Helix.Models/Extensions/ReleaseExtensions/ReleasedExtension.cs [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        [JsonProperty(PropertyName = "name")]
+        public string Name { get; protected set; }
+        [JsonProperty(PropertyName = "privacy_policy_url")]
+        public string PrivacyPolicyUrl { get; protected set; }
+        [JsonProperty(PropertyName = "request_identity_link")]
+        public bool RequestIdentityLink { get; protected set; }
+        [JsonProperty(PropertyName = "screenshot_urls")]
+        public string[] ScreenshotUrls { get; protected set; }
+        [JsonProperty(PropertyName = "state")]
+        public string State { get; protected set; }
```
Comment (Syzuna) [inline TwitchLib.Api.Helix.Models/Extensions/ReleaseExtensions/ReleasedExtension.cs:39]: string vs enum? thoughts?

File: TwitchLib.Api.Helix/Extensions.cs [hunk 4]
```diff
@@ -64,5 +65,25 @@ public Task<GetExtensionLiveChannelsResponse> GetExtensionLiveChannelsAsync(stri
         }
 
         #endregion
+
+        #region GetReleasedExtensions
+
+        public Task<GetReleasedExtensionsResponse> GetReleasedExtensionsAsync(string extensionId, string extensionVersion = null, string accessTokenOrApplicationAccessToken = null)
```
Comment (Syzuna) [inline TwitchLib.Api.Helix/Extensions.cs:71]: See previous commit review

File: TwitchLib.Api.Helix.Models/Bits/ExtensionBitsProducts/ExtensionBitsProduct.cs [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+        [JsonProperty(PropertyName = "sku")]
+        public string Sku { get; protected set; }
+        [JsonProperty(PropertyName = "cost")]
+        public Cost Cost { get; protected set; }
+        [JsonProperty(PropertyName = "in_development")]
+        public bool InDevelopment { get; protected set; }
+        [JsonProperty(PropertyName = "display_name")]
+        public string DisplayName { get; protected set; }
+        [JsonProperty(PropertyName = "expiration")]
+        public DateTime Expiration { get; protected set; }
```
Comment (Syzuna) [inline TwitchLib.Api.Helix.Models/Bits/ExtensionBitsProducts/ExtensionBitsProduct.cs:19]: Does DateTime parse RFC3339 reliably?

File: TwitchLib.Api.Helix/Bits.cs [hunk 6]
```diff
@@ -66,5 +67,19 @@ public Task<GetBitsLeaderboardResponse> GetBitsLeaderboardAsync(int count = 10,
         }
 
         #endregion
+
+        #region GetExtensionBitsProducts
+
+        public Task<GetExtensionBitsProductsResponse> GetExtensionBitsProductsAsync(bool shouldIncludeAll = false, string accessTokenOrApplicationAccessToken = null)
```
Comment (Syzuna) [inline TwitchLib.Api.Helix/Bits.cs:73]: See previous commits review

File: TwitchLib.Api.Helix/Bits.cs [hunk 7]
```diff
@@ -81,5 +81,14 @@ public Task<GetExtensionBitsProductsResponse> GetExtensionBitsProductsAsync(bool
         }
 
         #endregion
+
+        #region UpdateExtensionBitsProduct
+
+        public Task<UpdateExtensionBitsProductResponse> UpdateExtensionBitsProductAsync(ExtensionBitsProduct extensionBitsProduct, string accessTokenOrApplicationAccessToken = null)
```
Comment (Syzuna) [inline TwitchLib.Api.Helix/Bits.cs:87]: See previous reviews comments

File: TwitchLib.Api.Helix.Models/Streams/GetFollowedStreams/Stream.cs [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+        [JsonProperty(PropertyName = "id")]
+        public string Id { get; protected set; }
+        [JsonProperty(PropertyName = "user_id")]
+        public string UserId { get; protected set; }
+        [JsonProperty(PropertyName = "user_login")]
+        public string UserLogin { get; protected set; }
+        [JsonProperty(PropertyName = "user_name")]
+        public string UserName { get; protected set; }
+        [JsonProperty(PropertyName = "game_id")]
+        public string GameId { get; protected set; }
```
Comment (Syzuna) [inline TwitchLib.Api.Helix.Models/Streams/GetFollowedStreams/Stream.cs:19]: GameName is missing

File: TwitchLib.Api.Helix/Streams.cs [hunk 9]
```diff
@@ -159,6 +160,19 @@ public Task<GetStreamMarkersResponse> GetStreamMarkerAsync(string userId, string
 
             return TwitchGetGenericAsync<GetStreamMarkersResponse>("/stream/markers", ApiVersion.Helix, getParams, accessToken);
         }
+
+        public Task<GetFollowedStreamsResponse> GetFollowedStreamsAsync(string userId, int first = 100, string after = null, string accessToken = null)
+        {
+            var getParams = new List<KeyValuePair<string, string>>
+            {
+                new KeyValuePair<string, string>("user_id", userId),
+                new KeyValuePair<string, string>("first", first.ToString())
```
Comment (Syzuna) [inline TwitchLib.Api.Helix/Streams.cs:169]: checks for first boundries?

File: TwitchLib.Api.Helix/Extensions.cs (hunk 1 above)
Comment (swiftyspiffy) [reply TwitchLib.Api.Helix/Extensions.cs:48]: The point here was to differentiate between "regular" `accessToken` and one accepting both. The naming was intentional here. There is a single endpoint that I left with that name for the same reason.

File: TwitchLib.Api.Helix.Models/Extensions/ReleaseExtensions/Component.cs (hunk 2 above)
Comment (swiftyspiffy) [reply TwitchLib.Api.Helix.Models/Extensions/ReleaseExtensions/Component.cs:6]: Renamed

File: TwitchLib.Api.Helix.Models/Bits/ExtensionBitsProducts/ExtensionBitsProduct.cs (hunk 5 above)
Comment (swiftyspiffy) [reply TwitchLib.Api.Helix.Models/Bits/ExtensionBitsProducts/ExtensionBitsProduct.cs:19]: It does. A few models currently using it: GetStreams.Stream, CreateStreamMarker.CreatedMarker, Bits.Cheermote

File: TwitchLib.Api.Helix/Extensions.cs (hunk 4 above)
Comment (swiftyspiffy) [reply TwitchLib.Api.Helix/Extensions.cs:71]: Renamed

File: TwitchLib.Api.Helix/Bits.cs (hunk 6 above)
Comment (swiftyspiffy) [reply TwitchLib.Api.Helix/Bits.cs:73]: Renamed

File: TwitchLib.Api.Helix/Bits.cs (hunk 7 above)
Comment (swiftyspiffy) [reply TwitchLib.Api.Helix/Bits.cs:87]: Renamed

File: TwitchLib.Api.Helix.Models/Streams/GetFollowedStreams/Stream.cs (hunk 8 above)
Comment (swiftyspiffy) [reply TwitchLib.Api.Helix.Models/Streams/GetFollowedStreams/Stream.cs:19]: Copy paste error. Fixed.

File: TwitchLib.Api.Helix/Streams.cs (hunk 9 above)
Comment (swiftyspiffy) [reply TwitchLib.Api.Helix/Streams.cs:169]: Added

**Review by Syzuna — APPROVED** [review]
LGTM

**Pull request merged**